src/session.rs

```diff
@@ -332,7 +375,9 @@ impl <'a> Session <'a> {
   }
 
   pub fn send(&mut self, frame: Frame) -> Result<()> {
```

Comment:
I think this could be simplified a bit by just making `frame: Frame` in the method parameter into `mut frame: Frame` so you don't have to introduce the `mut_frame` local variable.